yt_dlp/plugins.py

```diff
@@ -44,7 +79,47 @@ def dirs_in_zip(archive):
         pass
     except Exception as e:
         write_string(f'WARNING: Could not read zip file {archive}: {e}\n')
-    return set()
+    return ()
+
+
+def external_plugin_paths():
+    def _get_package_paths(*root_paths, containing_folder):
+        for config_dir in orderedSet(map(Path, root_paths), lazy=True):
+            # We need to filter the base path added when running __main__.py directly
+            if config_dir == _BASE_PACKAGE_PATH:
+                continue
+            with contextlib.suppress(OSError):
+                yield from (config_dir / containing_folder).iterdir()
+
+    # Load from yt-dlp config folders
+    yield from _get_package_paths(
+        *get_user_config_dirs('yt-dlp'),
+        *get_system_config_dirs('yt-dlp'),
+        containing_folder='plugins',
+    )
+
+    # Load from yt-dlp-plugins folders
+    yield from _get_package_paths(
+        get_executable_path(),
+        *get_user_config_dirs(''),
+        *get_system_config_dirs(''),
+        containing_folder='yt-dlp-plugins',
+    )
+
+    yield from internal_plugin_paths()
+
+
+def internal_plugin_paths():
+    # Always try load from PYTHONPATH folders
```

Comment:
Why treat this differently than external_plugin_paths?

Reply:
I should probably add a comment for this.

https://github.com/yt-dlp/yt-dlp/pull/11305/commits/37aed891c464e5bf06cac703eb6dba8b192f30bd

iirc it is because if there are no external locations (due to no-external) (and no custom locations) then our PluginFinder is bypassed and instead python default finders(?) are used, which may have logic we don't want?